components/CategoryImage/CategoryImage.tsx

```diff
@@ -0,0 +1,37 @@
+'use client';
+
+import type { Category, TranslatedCategory } from '@prezly/sdk';
+import Image from '@prezly/uploadcare-image';
+import classNames from 'classnames';
+
+import { type CardSize, getCardImageSizes } from './lib';
+
+import styles from './CategoryImage.module.scss';
+
+type Props = {
+    category: Category;
+    translatedCategory: TranslatedCategory;
```

Comment:
I think these props could be simpler since you only need the `image` and `alt` props, so maybe just add those instead of passing the whole `category` and `translatedCategory`?

Reply:
Simpler in what sense? There's plenty of naming and chasing scalar props. I prefer to keep objects intact to match a higher level mental model.

Reply:
I guess in this context it doesn't matter too much since it's a server component but in client components using scalar props makes it easier for React to check for value equality and therefore less chance of a re-render.

It's also easier to test components like this if you can just mock a single value instead of mocking the entire object.
Also also easier to extract the component into a generic component later, if it's not coupled to a specific object.

So I guess the tl;dr is that I like to keep the interface as simple as possible, if possible. This component could have quite easily be just called `Image` without the need for `Category` and it would still work, if the props were changed. Just saying.

So agree, it's a matter of preference 😉

Reply:
Thanks for those points. Indeed my backend bias shows. Will keep them in mind for client components. :)